benchconnect/README.md

```diff
@@ -18,6 +18,28 @@ With [pipx](https://pypa.github.io/pipx/):
 pipx install benchconnect@git+https://github.com/conbench/conbench.git@main#subdirectory=benchconnect
 ```
 
+## Environment variables
+
+benchconnect relies on a number of environment variables to obtain various
+metadata:
+
+- `CONBENCH_URL`: The URL of the Conbench server. Required.
+- `CONBENCH_EMAIL`: The email to use for Conbench login. Only required if the
+server is private.
+- `CONBENCH_PASSWORD`: The password to use for Conbench login. Only required
+if the server is private.
+- `CONBENCH_REPOSITORY`: The URL of the repo being benchmarked. Defaults
+to `"https://github.com/apache/arrow"` if unset.
+- `CONBENCH_PR_NUMBER`: Integer of pull request number associated with
+the run being benchmarked.
+- `CONBENCH_COMMIT`: Hash of commit being benchmarked. If missing, will
+attempt to obtain it from `arrow::arrow_info()$build_info$git_id`, though
+this may not be populated depending on how Arrow was built.
+- `CONBENCH_HOST_NAME`: By default, the running machine host name will be
```

Comment:
Let's maybe comment on where/how this is after all used (e.g., "sent as value to the `machine-name` key when POSTing benchmark results" or so)?